packages/@react-aria/calendar/src/useCalendarCell.ts

```diff
@@ -209,13 +211,13 @@ export function useCalendarCell(props: AriaCalendarCellProps, state: CalendarSta
     },
     onPress() {
       // For non-range selection, always select on press up.
-      if (!('anchorDate' in state) && !state.isReadOnly) {
+      if ((!('anchorDate' in state) || isInvalid) && !state.isReadOnly) {
         state.selectDate(date);
         state.setFocusedDate(date);
       }
     },
     onPressUp(e) {
-      if (state.isReadOnly) {
+      if (state.isReadOnly || isInvalid) {
```

Comment:
I'm seeing some strange behavior when the user tries to select a new date range if the previous date range is invalid.
Reproduction:
1. Go to https://reactspectrum.blob.core.windows.net/reactspectrum/70944f51de68206e59a69852f87b6db2d7d1daf3/storybook/index.html?path=/story/date-and-time-daterangepicker--mindate-2010-1-1-maxdate-2020-1-1
2. Confirm an invalid range by entering one into the DateRangePicker (e.g. make it a range in 2022 like 4/12/2022 - 4/13/2022)
3. Open the DateRangePicker dialog
4. Navigate to December 2019.
5. Click and drag to select a range in the above month. Note that the color of the range as you drag is red (should it be? note that `isInvalid` is returning true for these press events even when selecting non-invalid dates)
6. Release your click to select your new range. Note that the dialog doesn't close and that if you move your mouse pointer the range is still following your mouse. No change event is fired indicating that a new range wasn't selected

I assume the behavior here was that the drag to select wasn't supposed to work in the first place? I can reproduce on all desktop browsers + iOS Safari. I confirmed in Android chrome that I wasn't able to drag to select the date range if the current date range was invalid.

Reply:
Great catch! I think I fixed it. Now we only prevent dragging the ends of an existing range, and not any other invalid dates. That was causing some weirdness. And also, we don't mark new ranges as invalid when you are in the middle of selecting one so it shouldn't be red anymore.